packages/hoppscotch-backend/src/user-collection/user-collection.service.ts

```diff
@@ -1141,14 +1145,24 @@ export class UserCollectionService {
       return E.left(USER_COLLECTION_CREATION_FAILED);
     }
 
+    // Fetch nested collections after transaction is committed
+    const importedCollectionsWithChildren: CollectionFolder[] = [];
+    for (const userCollection of userCollections) {
+      const exportedCollectionJSON =
+        await this.exportUserCollectionToJSONObject(userID, userCollection.id);
+      if (E.isLeft(exportedCollectionJSON))
+        return E.left(exportedCollectionJSON.left);
+      importedCollectionsWithChildren.push(exportedCollectionJSON.right);
+    }
```

Comment:
Is there a `N + 1` query concern arising here? Are we foreseeing potential timeouts for large collection exports?

```ts
const collectionIds = userCollections.map(c => c.id);
const exportedCollections = await this.exportMultipleUserCollectionsToJSONObject(
  userID,
  collectionIds
);
```

Reply:
@mirarifhasan, please ensure to track this separately.